strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java

```diff
@@ -220,6 +218,9 @@ private long writeArtifact(RepositoryPath repositoryPath,
         {
             // Store artifact digests in cache if we have them.
             addChecksumsToCacheManager(digestMap, repositoryPathId);
+            //
+            artifactEventListenerRegistry.dispatchEvent(
```

Comment:
why we need an event for this? 
I think it can be implemented directly in `ArtifactManagementService`.

Reply:
It should be flexible for each lauout provider to handle this event. Maven case may not be the same way handled like NPM for example. I suggest to leave it as it is.

Reply:
I think it will be the same. 

my teacher on the OOP at the institute said a very clever thing: “you don’t need to do that is not required of you”.

This is the only useful thing that I learned from him :smile:

Reply:
> I think it will be the same.

But we are not sure, and event approach is more flexible. 

@carlspring what do you think ? The thing is: we want to handle calculated checksums. In maven case, we will store them in the `.sha1` or `.md5` files. There might be other handlers for other layout providers, do you know some other case maybe ?

Reply:
NuGet uses `sha512`. I think it would be better to be more flexible here.

Reply:
afaik you already have layout specific checksums in `digestMap`

Reply:
also, events probably is not the best way to get flexibility here

Reply:
> also, events probably is not the best way to get flexibility here

On the contrary I believe it really is.

Reply:
> afaik you already have layout specific checksums in digestMap

AFAIS, currently `SHA1` and `MD5` are supported but it can be extended in the future.

I want to leave it as it is.

Reply:
please have a look at `LayoutFileSystemProvider.decorateStream` method, it provides layout specific `digestAlgorithmSet` for `LayoutOutputStream` you have.

Reply:
ok, now I understand :)

Done